### PR TITLE
Make necessary changes

### LINKED ISSUE #15: Update Address overview to fit on mobile
As a user, I want the address overview to be responsive, so that I can view it on my mobile. 

# Replication steps 

 1. In mobile view, click ‘Explore’ > ‘Transaction’

 2. Then click an address link in the `From` or To `column`

The address overview page has images overlapping with text and poor formatting. Please update it, to make it match the attached design.

# Prerequisites

You will need to know basic Javascript (jQuery) and css (scss) to do this task. The knowledge of nunjucks will be helpful.

# Guidelines & Hints

For the Transactions section at the end of the page, we always want to show the scrollbar, so the user knows in advance that there is more data to be seen. 

You can check the code used in the Menu -> Transactions page.

The address overview page is located at: views/pages/address-detail.njk.

The current scrollbar is implemented on views/pages/tx-list.njk (look for className scroll-bar-wrapper).

The code to handle the scrollbar is located at: src/assets/js/mobile/scroll-bar.js (you should not have to change this).

Add any other necessary mobile css to _mobile.scss to match the design.

Please make sure your code doesn’t affect the desktop version. 

We do not want to change the HTML structure. 

We also recommend putting all mobile related styling code to the `_mobile.scss` file. 

# Resources

## Repository link

https://github.com/cennznet/explorer/tree/master 

This ‘explorer’ repository contains several projects. 

The project you need to work on is in the ‘portal’ folder.

## Design:

Please refer to the InVision prototype here:  [UNcover Bounty Tasks](https://invis.io/7TUN9MQY259)

You will need to request access permission to access inspect mode on the InVision prototype. 

Screenshot of intended design also attached. 

*Please note there may be some inconsistencies with the InVision prototype and the live CSS. If this occurs, please use the styles defined by the exisiting CSS. 

## Get Up & Running 

Please check the README and CONTRIBUTING files:

https://github.com/cennznet/explorer/blob/master/portal/README.md https://github.com/cennznet/explorer/blob/master/portal/CONTRIBUTING.md 

# Test & Verify Your Changes 

## Mobile Devices 

Ensure your changes work on: 

- The latest Android and iPhones phones.
- Chrome, Safari and FireFox responsive views.

Ensure your changes work as expected and look exactly like the attached design. 

Also ensure that the changes do not introduce bugs or break the desktop version.

## Expected Outcome

**Given** user navigates to address overview page using mobile device

**When** user navigates to address section

**Then** user should see QR code with address

**And**  user should see full address

**And** user should see copy symbol

**Given** user navigates to address overview page using mobile device

**When** user navigates to address section

**And** user click the copy symbol

**Then** user should have address copied to the clipboard and a tooltip appears

**Given** user navigates to address overview page using mobile device

**When** user navigates to assets section

**Then** user should see all assets stored in that address

**Given** user navigates to address overview page using mobile device

**When** user navigates to assets section

**And** user click the buttons on the left top side of the assets section

**Then** user should see the assets displaying style changes

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see time

 **And** user should see address

 **And** user should see incoming

 **And** user should see outgoing

 **And** user should see type

  **And** user should see status

  **And** user should see balance

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

 **And** user clicks on arrow sign at the end of the line

**Then** user should see expanded section

 **And** user should see block

 **And** user should see transaction

 **And** user should see  size 

 **And** user should see  fee

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see transaction type selection dropdown

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see transaction flow selection dropdown

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see start date selection dropdown

**Given** user navigates to address overview page using mobile device

**When** user navigates to transactions section

**Then** user should see end date selection dropdown

# Current Screenshots

 
image

image

–––––––––––––––––––––––––––––––––––––––––––

# Designed Screen (how your design should look)
image

### LINKED ISSUE #17: Update Assets page to fit on mobile
As a user I want to see full assets page when I’m on my mobile.

# Replication steps 

1. Click 'Assets'  in mobile view

The assets page has poor formatting. Please update to make it match the attached design.

Please ensure: 

- All the columns can be seen, including: (symbol, volume, decimal,  type).

- The items in the ‘Type’ dropdown are visible.

Currently a different ‘Asset’ and/or ‘Type’ dropdown item can be set and the results can be filtered when the ‘Filter’ button is clicked.

# Prerequisites

You will need to know basic Javascript (jQuery) and css (scss) to do this task.

# Resources

## Repository link

https://github.com/cennznet/explorer/tree/master 

This ‘Explorer’ repository contains several projects. 

The project you need to work on is in the ‘portal’ directory.

# Design

Please refer to the InVision prototype here:   [UNcover Bounty Tasks](https://projects.invisionapp.com/share/7TUN9MQY259#/screens/391426174)

You will need to request access permission to access inspect mode on the InVision prototype. 

Screenshot of intended design also attached. 

*Please note there may be some inconsistencies with the InVision prototype and the live CSS. If this occurs, please use the styles defined by the exisiting CSS. 

# Get Up & Running

Please check the README and CONTRIBUTING files:

https://github.com/cennznet/explorer/blob/master/portal/README.md https://github.com/cennznet/explorer/blob/master/portal/CONTRIBUTING.md 

# Guidelines & Hints

The main reason why the information looks poor on mobile is because it was coded up using an HTML Table Tag. 

We do not want to change the HTML structure. 

The main thing to do here is to add a horizontal scrollbar and make the table data scrollable.

We also want to always show the scrollbar, so the user knows in advance that there is more data to be seen.

The assets page is located at: views/pages/token-list.njk.

The current scrollbar is implemented on views/pages/tx-list.njk (look for className scroll-bar-wrapper).

The code to handle the scrollbar is located at: src/assets/js/mobile/scroll-bar.js (you should not have to change this).

Add any other necessary mobile css to _mobile.scss to reflect the design (hint: see className mobile-wrapper-tx-table.

The transaction’s page (pages/tx-list.njk) filter controls have already been styled so please refer there for   guidance on how to style the assets page’s filter controls.

# Test & Verify Your Changes 

## Mobile Devices 

Ensure your changes work on:

- The latest Android and iPhones phones.

- Chrome, Safari and FireFox responsive views.

Ensure your changes work as expected and look exactly like the attached design.

Also ensure that the changes do not introduce bugs or break the desktop version.

# Expected Outcome 

Given user navigates to assets page using mobile device 

When user navigates to the assets section 

Then user should see symbol 

And user should see volume

And user should see decimal

And user should see type

Given  user navigates to assets page using mobile device 

When user navigates to the assets filter section 

And user clicks on Assets or Type filter 

Then user should see native scrollbar 

Given user navigates to assets page using mobile device 

When user navigates to the assets filter section 

And user selects filter 

Then user should see filtered results 

Given user navigates to assets page using desktop 

When user navigates to the assets section 

Then user should see standard full screen UI

## If you have any questions, please don’t hesitate to contact our team on [Slack](https://join.slack.com/t/centralitydev/shared_invite/enQtNjk5NjE1MDI5NjgzLWJlOTM2NjMxNTMyZDY0OGM0MjY2YjQ1YjJkNTk2ZTIzNTFmYmM3OTIwZWE3NzNhMzE4Mjg4MmY1YzBiZDk0ZDE).

Current screenshots
image

–––––––––––––––Designed Screen (how your design should look)
Assets

### PR DESCRIPTION
**Pull request by iamonuwa**
Fixes #15

### CODE REVIEW

File: portal/src/assets/style/_mobile.scss [hunk 1]
```diff
@@ -2,544 +2,605 @@
...
+			border-radius: 5px;
+			border: 2px solid #ffffff;
+		}
+		&::-webkit-scrollbar-track {
+			background-color: #ececec;
+			border-radius: 3px;
+		}
+	}
+	display: none;
 }
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: This block of code should not of changed. Please use the same indentation and formatting  of project.  You have not added any code to the  .scroll-bar-wrapper block

File: portal/src/assets/style/_mobile.scss [hunk 2]
```diff
@@ -2,544 +2,605 @@
...
+		margin: 5px;
+
+		&::-webkit-scrollbar {
+			-webkit-appearance: none;
+		}
+		&::-webkit-scrollbar:horizontal {
+			height: 17px;
+		}
+		&::-webkit-scrollbar-thumb {
+			background-color: rgba(0, 0, 0, 0.5);
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: 0.5 has changed, it was 0.5 , (  I think you don't need to change this )

File: portal/src/assets/style/_mobile.scss [hunk 3]
```diff
@@ -2,544 +2,605 @@
...
+			border-radius: 5px;
+			border: 2px solid #ffffff;
+		}
+		&::-webkit-scrollbar-track {
+			background-color: #ececec;
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: the color code was #ECECEC, i guess you don't need to change this

File: portal/src/assets/style/_mobile.scss [hunk 4]
```diff
@@ -2,544 +2,605 @@
...
+		padding: 2rem 0;
+		line-height: 2.5rem;
+	}
+
+	.footer-content {
+		flex-direction: column;
+	}
+
+	h1,
+	.h1 {
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: Please use the same indentation and formatting

File: portal/src/assets/style/_mobile.scss [hunk 5]
```diff
@@ -2,544 +2,605 @@
...
+				padding-top: 1.7rem;
+				font-weight: 600;
+
+				.tx-value {
+					font-size: 20px;
+					white-space: nowrap;
+				}
+
+				.tx-date {
+					color: #1e2022;
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: Please do not change things you don't have to, the color can stay as uppercase

File: portal/src/assets/style/_mobile.scss [hunk 6]
```diff
@@ -2,544 +2,605 @@
...
+			margin-right: 10px;
+		}
+
+		.content {
+			padding: 0.5rem 0;
+			display: inline;
+			flex: 3;
+			white-space: normal;
+			word-break: break-word;
+			color: #1e2022;
```
Comment (centrality-bounties) [inline portal/src/assets/style/_mobile.scss]: Please do not change things you don't have to, the color can stay as uppercase

File: portal/views/partials/address/address-info.njk [hunk 7]
```diff
@@ -13,3 +13,19 @@
     </div>
   </div>
 </div>
+
+<div class="address-info block-section mobile-only">
+<div class="qr-img">
```
Comment (centrality-bounties) [inline portal/views/partials/address/address-info.njk]: please indent this div, and use correct indentation

**Comment by iamonuwa** [comment]
@centrality-bounties @AlexZhenWang please take a look

File: portal/views/partials/address/address-tokens.njk [hunk 8]
```diff
@@ -1,4 +1,4 @@
-<div class="block-section address-tokens">
+<div class="block-section address-tokens isDesktop">
```
Comment (centrality-bounties) [inline portal/views/partials/address/address-tokens.njk]: we already have a test for  isDesktop , if you have to use this css class  use underscore syntax like the rest of the css. i.e: is_desktop

File: portal/views/partials/address/address-tokens.njk [hunk 9]
```diff
@@ -69,3 +69,22 @@
     </div>
   </div>
 </div>
+
+
+<div class="block-section address-tokens isMobile">
```
Comment (centrality-bounties) [inline portal/views/partials/address/address-tokens.njk]: we already have a test for isMobile , if you have to use this css class use underscore syntax like the rest of the css. i.e: is_mobile

File: portal/views/partials/address/address-tokens.njk [hunk 10]
```diff
@@ -69,3 +69,22 @@
...
+  <div class="block-section__header--cols">
+	<h2>{{ _('navbar.assets') }} ({{ address.balances.length }} {{ _('helper.field.total') }})</h2>
+  </div>
+  <div class="block-section__content filter-form">
+  <select id="txToken" name="txTokens" class="selectpicker" data-width="100%">
```
Comment (centrality-bounties) [inline portal/views/partials/address/address-tokens.njk]: you gave this an ID but it is not used, please remove it or use it

**Comment by iamonuwa** [comment]
@centrality-bounties @AlexZhenWang made changes to reflect the invision design. 2 issues are remaining which I need guidance on how to go about it.

1. I tried creating some i18n value but anytime I reload the app, it removes it and create a string of same value for me.
2. The invision design had the select token dropdown to have the token symbol and token balance. I tried using the `balanceInfo.assetSymbol` to retrieve the symbol but the app fails anytime that happens. What's the solution to this please?

**Comment by centrality-bounties** [comment]
Hello, @iamonuwa iamonuwa 
Check the desktop version to see how the token symbol and token balance are used.  You can see that an html item is created correctly for them, just do it a similar way.

The Asset drop down can be customised using data-subtext like:
<option className=“what_you_want_to_style_this” data-subtext=“right”>  left
 </option> see https://stackoverflow.com/questions/46172170/select-dropdown-with-customized-option-labels and https://developer.snapappointments.com/bootstrap-select/examples/#style-individual-options.

Double check how other i18n values are used, like :  navbar.filter.transactions 
I @iamonuwa iamonuwa 
Check the desktop version to see how the token symbol and token balance are used.  You can see that an html item is created correctly for them, just do it a similar way.

The Asset drop down can be customised using data-subtext like:
<option className=“what_you_want_to_style_this” data-subtext=“right”>  left
 </option> see https://stackoverflow.com/questions/46172170/select-dropdown-with-customized-option-labels and https://developer.snapappointments.com/bootstrap-select/examples/#style-individual-options.

Double check how other i18n values are used, like :  navbar.filter.transactions 

Thanks Nick

**Comment by iamonuwa** [comment]
> Hello, @iamonuwa iamonuwa
> Check the desktop version to see how the token symbol and token balance are used. You can see that an html item is created correctly for them, just do it a similar way.
> 
> The Asset drop down can be customised using data-subtext like:
> 
> ```
> <option className=“what_you_want_to_style_this” data-subtext=“right”>  left
>  </option>
> ```
> 
> see https://stackoverflow.com/questions/46172170/select-dropdown-with-customized-option-labels and https://developer.snapappointments.com/bootstrap-select/examples/#style-individual-options.
> 
> Double check how other i18n values are used, like : navbar.filter.transactions
> 
> Thanks Nick

Thanks Nick. Succeded with formatting the dropdown. But the translation still doesn't work.

**Comment by iamonuwa** [comment]
Still failing for me. @centrality-bounties @AlexZhenWang

**Comment by iamonuwa** [comment]
@centrality-bounties @AlexZhenWang fixed the i18n error. For the select option onChange event, I may not be able to fix that without adding a button to the UI.

**Comment by iamonuwa** [comment]
Closing this infavor of #

**Pull request closed without merging**